### PR TITLE
Adds compatibility to get docker image from non-default registry

### PR DESCRIPTION
**Pull request by mahdipub**
Enables the pipeline to fetch from dockerhub and other registries Infra issue 9128

Signed-off-by: Mahdi Ardekanian <mahdi@ibm.com>

### CODE REVIEW

**Comment by AdamBrousseau** [comment]
jenkins compile alinux,xlinux,zlinux jdk11

File: buildenv/jenkins/common/build.groovy [hunk 1]
```diff
@@ -771,8 +781,9 @@ def prepare_docker_environment() {
     // if there is no id found then attempt to pull it
     if (!DOCKER_IMAGE_ID) {
         echo "${DOCKER_IMAGE} not found locally, attempting to pull from dockerhub"
-        dockerCredentialID = variableFile.get_user_credentials_id('dockerhub')
-        docker.withRegistry("", "${dockerCredentialID}") {
+        dockerRegistery = getDockerRegistery(DOCKER_IMAGE)
```
Comment (AdamBrousseau) [inline buildenv/jenkins/common/build.groovy:784]: Registry is spelled wrong

**Comment by AdamBrousseau** [comment]
jenkins compile xlinux jdk11

File: buildenv/jenkins/common/build.groovy (hunk 1 above)
Comment (mahdipub) [reply buildenv/jenkins/common/build.groovy]: Resolved.

**Comment by AdamBrousseau** [comment]
Builds are passed the point of the change. Relaunched the xlinux and targeted a machine that didn't already have the matching container in order to force the pull and exercise the code.
z (no docker) https://openj9-jenkins.osuosl.org/job/Build_JDK11_s390x_linux_Personal/855/
a (docker no pull) https://openj9-jenkins.osuosl.org/job/Build_JDK11_aarch64_linux_Personal/683/
x (docker with pull) https://openj9-jenkins.osuosl.org/job/Build_JDK11_x86-64_linux_Personal/795/

**Review by AdamBrousseau — APPROVED** [review]

**Pull request merged**